src/posix/platform/platform-posix.h

```diff
@@ -442,6 +442,20 @@ void platformUdpUpdateFdSet(otInstance *aInstance, fd_set *aReadFdSet, int *aMax
  */
 int SocketWithCloseExec(int aDomain, int aType, int aProtocol);
 
+/**
+ * This function creates a socket with SOCK_CLOEXEC and SOCK_NONBLOCK flag set.
+ *
+ * @param[in]   aDomain     The communication domain.
+ * @param[in]   aType       The semantics of communication.
+ * @param[in]   aProtocol   The protocol to use.
+ *
+ * @returns The file descriptor of the created socket.
+ *
+ * @retval  -1  Failed to create socket.
+ *
+ */
+int NonBlockSocketWithCloseExec(int aDomain, int aType, int aProtocol);
```

Comment:
Is it possible to merge into SocketWithCloseExec() by adding an additional argument for nonblocking?